src/main/java/uk/nhs/tis/sync/job/PersonRecordStatusJob.java

```diff
@@ -111,83 +93,21 @@ private LocalDate validateDateParamFormat(String dateStr) throws DateTimeParseEx
     return LocalDate.parse(dateStr);
   }
 
-  protected String getJobName() {
-    return this.getClass().getSimpleName();
-  }
-
-  protected int getPageSize() {
-    return DEFAULT_PAGE_SIZE;
-  }
-
   protected EntityManagerFactory getEntityManagerFactory() {
     return entityManagerFactory;
   }
 
-  protected List<Long> collectData(long lastPersonId, String queryString,
-      EntityManager entityManager) {
-    Query query =
-        entityManager.createNativeQuery(queryString).setParameter("lastPersonId", lastPersonId);
-
-    List<BigInteger> resultList = query.getResultList();
-    return resultList.stream().filter(Objects::nonNull)
-        .map(result -> Long.parseLong(result.toString()))
-        .collect(Collectors.toList());
-  }
-
-  protected int convertData(Set<Person> entitiesToSave, List<Long> entityData,
-      EntityManager entityManager) {
-    int entities = entityData.size();
-    entityData.stream().map(id -> entityManager.find(Person.class, id))
-        .filter(Objects::nonNull)
-        .filter(p -> p.getStatus() != p.programmeMembershipsStatus())
-        .forEach(p -> {
-          p.setStatus(p.programmeMembershipsStatus());
-          entitiesToSave.add(p);
-        });
-    return entities - entitiesToSave.size();
-  }
-
-  @Autowired(required = false)
-  private ApplicationEventPublisher applicationEventPublisher;
-
-  @ManagedOperation(description = "Is the sync job currently running")
-  public boolean isCurrentlyRunning() {
-    return mainStopWatch != null;
-  }
-
-  @ManagedOperation(description = "The current elapsed time of the current sync job")
-  public String elapsedTime() {
-    return mainStopWatch != null ? mainStopWatch.toString() : "0s";
+  protected String getDateOfChangeOverride() {
+    return dateOfChangeOverride;
```

Comment:
I'll leave it up to you to decide if it is more readable but you can specify a `protected String dateOfChangeOverride;` in the super class and override it here by changing this class to set it:
```java
  @Value("${application.jobs.personRecordStatusJob.dateOfChangeOverride}")
  public void setDateOverride(String dateOfChangeOverride) {
    this.dateOfChangeOverride = dateOfChangeOverride; 
  }
```

Reply:
Thanks Pepe, putting it in superclass is more readable, and there's another attribute `entityManagerFactory` which can be moved to superclass too I think.